src/XrdSys/XrdSysE2T.cc

```diff
@@ -63,6 +63,12 @@ int initErrTable()
            lastGood = i;
           }
       }
+   // NOTE: On systems without EAUTH (authentication error; currently all Glibc systems but GNU Hurd),
+   // EAUTH is remapped to EBADE ('invalid exchange').  Given there's no current XRootD use of a
+   // syscall that can return EBADE, we assume EBADE really means authentication denied.
+#if defined(EBADE)
+   Errno2String[EBADE] = "authentication failed";
+#endif
```

Comment:
Another minor note. Is the assumption that `EBADE` == authentication failed reasonable? Can something else be mistaken for authentication failure because of this? 

Checking a bit, I also figured out that macOS has `EAUTH` but not `EBADE`, and on Linux musl doesn't seem to have either. I wonder what happens in that case.

Reply:
(Indeed, MacOS lacks `EBADE`, that's why I had to do the `#if` macro)

> Is the assumption that EBADE == authentication failed reasonable? Can something else be mistaken for authentication failure because of this?

I think it's reasonable but not necessarily bulletproof.  The EBADE response is pretty esoteric and I couldn't spot any syscalls used that would trigger it on MacOS.

Can I guarantee that this remains true in the future?  Of course not.  My feeling is this would lead to an improvement in 99.9% of cases, however.